query.go

```diff
@@ -53,15 +61,20 @@ func (dht *IpfsDHT) newQuery(k string, f queryFunc) *dhtQuery {
 	}
 }
 
+// QueryPathFunc is a function that holds a path's state in its closure.
+// It returns the QueryFunc for one particular path. Its parameter, `path`,
+// specifies the path index number from 0 to numPaths - 1
+type queryPathFunc func(pathIndex int, numPaths int) QueryFunc
+
 // QueryFunc is a function that runs a particular query with a given peer.
 // It returns either:
 // - the value
 // - a list of peers potentially better able to serve the query
 // - an error
-type queryFunc func(context.Context, peer.ID) (*dhtQueryResult, error)
+type QueryFunc func(context.Context, peer.ID) (*dhtQueryResult, error)
```

Comment:
I think this should be private. `queryFunc`.

Reply:
Sure